### ISSUE TITLE
Help me understand that the strength point mention in the paper is infact the weakness of the paper? 

### ISSUE THREAD

**Issue by nile649** [srebuffi/lsd-clusters#2]
I haven’t fully grasped the paper, but the idea of having pair-wise new labels within the minibatch is, in fact, the limiting factor of the algo.
Most of the images can only be reduced to 512x512 without losing much information, and passing Image of size 512x512 with mini-batch over 32 will result in a computational overflow.
correct me If I am wrong?

The color is another feature that often results in bias, using rot-net we may reduce some geometrical bias but still, there is no mention of the color. 

So is the paper just a proof of concept?

**Comment by nile649** [comment]
Also, help me here,
VGG features trained on place 365+Image net -> (GAP / or aggregator ) -> cosine similarity = cluster already works great, but here I don't see anything related to it. Any idea in theory how your method will compare to the above pipeline.

**Comment by hyenal** [comment]
Hi @nile649 thanks for your interest in our work :)

There area few questions so I will try to answer them inline!

>I haven’t fully grasped the paper, but the idea of having pair-wise new labels within the minibatch is, in fact, the limiting factor of the algo.
>Most of the images can only be reduced to 512x512 without losing much information, and passing Image of size 512x512 with mini-batch over 32 will result in a computational overflow.
>correct me If I am wrong?

We directly operate on the network embedded space. Hence images are transformed to 512-d vectors before computing pairwise similarity which is a scalar. While building the distance matrix grows quadratically with the batch size, in your example I am struggling to see where is the computational overflow since it will results in building a 32x32 matrix.
If you think the limiting factor is in fact using a neural network to compute the embedding, as recalled in appendix A of the paper, after pre training we freeze the three first blocks of the network. So you can pre compute embedding if needs be.

>The color is another feature that often results in bias, using rot-net we may reduce some geometrical bias but still, there is no mention of the color.

Finding a good representation of the image is a hard problem. We are aware that other work such as IIC (Ji et al.) also uses Sobel filter to extract edges of the image, but other work you cite using pre-trained VGG works on raw images.
Color as such gives you more information which can be useful. 
If you were thinking of some work in particular that mentioned using color is problematic I am happy to discuss :)


>VGG features trained on place 365+Image net -> (GAP / or aggregator ) -> cosine similarity = cluster already works great, but here I don't see anything related to it. Any idea in theory how your method will compare to the above pipeline.

Using feature extracted from VGG (or ResNet 50 for VaDE) can be seen as an upper bound for most of the benchmark used. In fact for most of them they share all their classes with Imagenet.
In the example you mentioned could you indicate the results you (or the paper you are quoting) obtain ? 
In fact what data you pre-trained your network on will likely bias your representation. There are a few work that shows in semi supervised learning ([Zhou et al.](https://arxiv.org/pdf/1812.05313.pdf), [Oliver et al. (see fig.2 and accompanying text)](https://arxiv.org/pdf/1804.09170.pdf)) how transferring from unrelated data can actually hurt performances.

**Comment by nile649** [comment]
Thanks for the clarification especially for the first and the last answer.
